controller/src/controller/main_systems/system_monitor.py

```diff
@@ -304,6 +311,8 @@ async def _handle_comm_from_instrument_comm(self) -> None:
                         )
                 case {"command": "get_metadata", **metadata}:
                     system_state_updates["instrument_metadata"] = metadata
+                    # immediately check connection status
+                    await self._queues["to"]["instrument_comm"].put({"command": "check_connection_status"})
```

Comment:
You don't necessarily need to send this command from here, you could have instrument comm send this command itself when it receives the get_metadata response